### DISCUSSION TITLE
Iperf3 measures throughput with UDP

### DISCUSSION THREAD

**Discussion by xxshijie** [esnet/iperf#1284] [Q&A]
When iperf3 uses UDP to measure throughput, use the -b parameter to set the bandwidth larger than the actual bandwidth in the network, such as a 10 Mbps link is configured to send data at 100Mbps, is the client sending data at 100Mbps? If yes, is the packet loss rate very large? How does it measure bandwidth?

**Comment by bmah888** [comment]
Clearly there's no way to send 100Mbps on a 10Mbps link. Because UDP doesn't guarantee the delivery of data, the packets that don't fit onto the link will likely just get dropped, probably in the kernel on the sending host due to lack of buffer space. In the setting you described, yes the packet loss rate will be very large (assuming the sending host is capable of sending a 100Mbps stream, most can).

There are a couple of ways to measure the bandwidth...one is to adjust the sending rate with the `-b` parameter until you find something that doesn't result in any significant packet loss.

Another way is to use TCP instead of UDP...because TCP guarantees reliable, in-order delivery, it will only send as fast as the network is capable of supporting (in its steady state).

**Reply by vladannikodijevic** [reply]
hi there,
is there a way to see the statistics on TCP test in iPerf? What would be the easiest way?

**Reply by davidBar-On** [reply]
Hi, when you run iperf3 the statistics are displayed.  To see more detailed statistics in JSON format use the `-J` options.
